05-angularjs/js/app.js

```diff
@@ -0,0 +1,165 @@
+var app = angular.module('moviesApp', ['ngRoute']); 
+
+app.config(['$routeProvider', function ($routeProvider) {
+    $routeProvider.
+    	when('/', {
+        	templateUrl: 'views/main.html',
+        	controller: 'moviesController'
+      	}).
+      
+      	when('/movie-description/:movieId', {
+        	templateUrl: 'views/movie-description.html',
+        	controller: 'descriptionController'
+      	}).
+      	
+      	otherwise({
+        	redirectTo: 'views/main.html'
+      	});
+}]); 
+
+/* -------------------------------------------- MOVIES CONTROLLER ---------------------------------------------------------- */
+app.controller('moviesController', ['$scope', 'movieService',  function ($scope, movieService) { 
+
+	$scope.listMovies = movieService.getMovies();
+	$scope.editPanel = false;
+	$scope.movieIndex = null;
+		
+	$scope.showEditPanel = function(index) {
+		if (!$scope.editPanel) {
+
+			$scope.editPanel = true;
+			$scope.movieIndex = index;
+			var currentMovie = $scope.listMovies[index];
+
+			var movieToEdit = {
+				name: currentMovie.name,
+				director: currentMovie.director,
+				image: currentMovie.image,
+				description: currentMovie.description
+			};
+
+			$scope.movieEdit = movieToEdit;
+			scrollToMenu();
+
+		} else {
+
+			$scope.editPanel = false;
+			$scope.movieIndex = null;
+		}
+	};
+
+	$scope.addMovie = function() {
+
+		$scope.movieNew.image = "imgs/new.jpg";
+		$scope.listMovies.unshift($scope.movieNew);
+
+		$scope.movieNew = {};
+		$scope.editPanel = false;
+		$scope.toggle = !$scope.toggle;
+	};
+
+	$scope.removeMovie = function(index) {
+		
+		$scope.listMovies.splice(index, 1);
+	};		
+
+	$scope.editMovie = function() {
+
+		$scope.listMovies[$scope.movieIndex] = $scope.movieEdit;
+		$scope.movieEdit = {};
+		$scope.editPanel = false;
+	};
+
+	$scope.showMovie = function() {
+
+		var movies = [];
+
+		for(var i = 0; i <  $scope.listMovies.length; i++) {
+
+			var iMovie = {
+				name:  $scope.listMovies[i].name,
+				director: $scope.listMovies[i].director,
+				image: $scope.listMovies[i].image,
+				description: $scope.listMovies[i].description
+			};
+
+			movies.push(iMovie);
+		}
+
+		movieService.saveMovies(movies);
```

Comment:
Maybe updating the dataset would be better in another place(s), like edit add or remove movies.